airflow/version.py

```diff
@@ -24,6 +24,13 @@
 except ImportError:
     import importlib_metadata as metadata
 
-version = metadata.version('apache-airflow')
+try:
+    version = metadata.version('apache-airflow')
+except metadata.PackageNotFoundError:
+    import logging
+
+    log = logging.getLogger(__name__)
+    log.warning("Package metadata could not be found. Overriding it with version found in setup.py")
+    from setup import version
```

Comment:
Is this logging necessary? Is there a possibility that the version may differ?

Reply:
This is actually a protection against the case when you think you have airflow installed from the package (i.e. in production) but you actually don't (for example you have it installed with `pip -e .`. I think all production installations should be installed from packages (this is what I am working on in #12685. So the warning is deliberate here to be warned if that's the case.

Reply:
Better safe than sorry.